18-websockets/classworks/Alexey_Goremykin/src/pages/Chat/Chat.jsx

```diff
@@ -1,65 +1,76 @@
-import React, { useState } from 'react';
+import React, { useState, useCallback, useEffect } from 'react';
 import './Chat.css';
 import Message from './components/Message';
 import io from 'socket.io-client';
-const socket = io('http://localhost:3001/', {transports: ['websocket']});
 
-socket.on('connection', () => {
-  console.log(socket.connected);
-});
+const socket = io('http://localhost:3001/');
 
 function Chat() {
-
   const [message, setMessage] = useState('');
   const [userName, setUserName] = useState('');
   const [messages, setMessages] = useState([]);
-  const [typingMessage, setTypingMessage] = useState('')
+  const [typingMessage, setTypingMessage] = useState([]);
 
-  const sendMessage = () => {
-    socket.emit('send-message', {userName, message});
-    socket.on('new-message', ([...newMessages]) => {
-      setMessages(newMessages);
-    })
-  }
+  const setTypingTimeout = () => setTimeout(() => { setTypingMessage([]) }, 2000);
```

Comment:
you will remove all typing users when time is out. I think you should handle it for every user.